tools/buildmgr/cbuildgen/config/IAR.8.50.6.cmake

```diff
@@ -95,25 +98,62 @@ function(cbuild_set_defines lang defines)
   set(${defines} ${TMP_DEFINES} PARENT_SCOPE)
 endfunction()
 
-set(ASM_DEFINES ${DEFINES})
-cbuild_set_defines(ASM ASM_DEFINES)
+set(OPTIMIZE_VALUES    "none" "balanced" "size" "speed")
+set(OPTIMIZE_CC_FLAGS  "-On"  "-Oh"      "-Ohz" "-Ohs")
+set(OPTIMIZE_CXX_FLAGS ${OPTIMIZE_CC_FLAGS})
 
-set(CC_DEFINES ${DEFINES})
-cbuild_set_defines(C CC_DEFINES)
+set(DEBUG_VALUES       "on"      "off")
+set(DEBUG_ASM_FLAGS    "-r"      "")
+set(DEBUG_CC_FLAGS     "--debug" "")
+set(DEBUG_CXX_FLAGS    ${DEBUG_CC_FLAGS})
 
-if(BYTE_ORDER STREQUAL "Little-endian")
-  set(IAR_BYTE_ORDER "little")
-elseif(BYTE_ORDER STREQUAL "Big-endian")
-  set(IAR_BYTE_ORDER "big")
-endif()
+set(WARNINGS_VALUES    "on" "off")
+set(WARNINGS_ASM_FLAGS ""   "-w-")
+set(WARNINGS_CC_FLAGS  ""   "--no_warnings")
+set(WARNINGS_CXX_FLAGS ${WARNINGS_CC_FLAGS})
+set(WARNINGS_LD_FLAGS  ${WARNINGS_CC_FLAGS})
 
+function(cbuild_get_option_flags lang option value flags)
+  if(NOT DEFINED ${option}_${lang}_FLAGS)
+    return()
+  endif()
+  list(FIND ${option}_VALUES "${value}" _index)
+  if (${_index} GREATER -1)
+    list(GET ${option}_${lang}_FLAGS ${_index} flag)
+    set(${flags} "${flag} ${${flags}}" PARENT_SCOPE)
+  else()
+    string(TOLOWER "${option}" _option)
+    message(FATAL_ERROR "unkown "${_option}" value '${value}' !")
+  endif()
+endfunction()
+
+function(cbuild_get_options_flags lang optimize debug warnings flags)
+  set(opt_flags)
+  cbuild_get_option_flags(${lang} OPTIMIZE "${optimize}" opt_flags)
+  cbuild_get_option_flags(${lang} DEBUG    "${debug}"    opt_flags)
+  cbuild_get_option_flags(${lang} WARNINGS "${warnings}" opt_flags)
+  set(${flags} "${opt_flags} ${${flags}}" PARENT_SCOPE)
+endfunction()
```

Comment:
Can we make the functions names consistent with `cbuild_set_defines`?
Maybe this functions should be called `set` instead of `get`? It **gets** the generic symbols but the goal is to **set** the options flags, does it make sense?